plutus-tx/src/PlutusTx/Blueprint/Class.hs

```diff
@@ -0,0 +1,84 @@
+{-# LANGUAGE AllowAmbiguousTypes   #-}
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE KindSignatures        #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE TypeApplications      #-}
+
+module PlutusTx.Blueprint.Class where
+
+import Prelude
+
+import Data.Kind (Type)
+import PlutusTx.Blueprint.Schema (Schema (..))
+import PlutusTx.Builtins (BuiltinByteString, BuiltinData, BuiltinString)
+
+class HasDataSchema (t :: Type) (canReferTypes :: [Type]) where
```

Comment:
haddock? what is this? I can't read your mind :)

Reply:
Sure, will add.